src/modules/LanItems.ycp

```diff
@@ -1831,28 +1832,49 @@ global map FindMatchingDevice(map interface) {
     return tosel;
 }
 
-global void DeleteItem(){
- y2milestone("deleting ... %1", Items[current]:$[]);
- string ifcfg = Items[current, "ifcfg"]:"";
- string hwcfg = Items[current, "hwcfg"]:"";
+global void DeleteItem()
+{
+    y2milestone("deleting ... %1", Items[current]:$[]);
 
- if (IsNotEmpty(ifcfg)) {
-	 NetworkInterfaces::Delete(ifcfg);
-	 NetworkInterfaces::Commit();
-	 Items[current, "ifcfg"] = "";
-	}
- if (! (size(Items[current, "hwinfo"]:$[])>0) ){
-  map<integer, any> tmp_items = $[];
-  foreach(integer key, any value, Items, {
-   if (key==current) continue;
-    else{
-	if (key<current) tmp_items[key]=Items[key]:$[];
-		else tmp_items[key-1]=Items[key]:$[];
-	}
-  });
-  Items = tmp_items;
- }
- SetModified();
+    string ifcfg = Items[current, "ifcfg"]:"";
+    string hwcfg = Items[current, "hwcfg"]:"";
+
+    if (IsNotEmpty(ifcfg))
+    {
+        NetworkInterfaces::Delete(ifcfg);
+
+        // this depends on above NetworkInterfaces::Delete
```

Comment:
hm, I've overlooked the ```Select``` in ```GetValue```, so the comment is useless.

No problem to add another level of ```if``` is no problem, so I'll adapt it.